contrib/masterfiles-stage/masterfiles-stage.sh

```diff
@@ -111,93 +31,7 @@ git_tag_or_commit_masterstage() {
-
-  # If we don't have a git clone wipe the directory and create a new clone and
-  # ensure the proper branch/tag is checked out.
-  else
-      rm -rf "${TMP}"/* "${TMP}"/.??*
-      git clone --no-hardlinks "${GIT_URL}" "${TMP}" && cd "${TMP}" && git checkout "${GIT_TAG_OR_COMMIT}" || error_exit "Failed to stage '${GIT_TAG_OR_COMMIT}' from '${GIT_URL}'"
-  fi
-
-  if /var/cfengine/bin/cf-promises -T "${TMP}"; then
-    if ! /var/cfengine/bin/cf-promises -cf "${TMP}/update.cf"; then
-        error_exit "Update policy staged in ${TMP} could not be validated, aborting."
-    fi
-      # roll out the release if the release IDs are different
-#      if /usr/bin/diff -q "${TMP}/cf_promises_release_id" "${MASTERDIR}/cf_promises_release_id" ; then
-#          #echo "No release needs to be made, the release IDs are the same."
-#          touch "${TMP}"
-#      else
-          cd "${TMP}" && (
-          chown -R root:root "${TMP}" && rsync -rltDE -c --delete-after --chmod=u+rwX,go-rwx "${TMP}/" "${MASTERDIR}/" && echo "Successfully staged a policy release of '${GIT_TAG_OR_COMMIT}' from '${GIT_URL}' to '${MASTERDIR}' on $(date)"
-      )
-#      fi
-  else
-    error_exit "The staged policies in ${TMP} could not be validated, aborting."
-  fi
-}
-
-svn_branch() {
-# Contributed by John Farrar
-
-    if ! type "svn" >/dev/null ; then
-	error_exit "svn not found on path: ${PATH}"
-    fi
-
-    CHECKSUM_FILE="svn_promise_checksums"
-
-    # We probably want a different temporary location for each remote repository
-    # so that we can avoid conflicts and potential confusion.
-    # Example: 
-    # ROOT="/opt/cfengine/masterfiles_staging"
-    # PARAMS="/var/cfengine/policychannel/production_1.sh"
-    # TMP=/opt/cfengine/masterfiles/staging/_tmp_var_cfengine_policychannel_production_1_sh
-
-    TRANSLATED_PARAMS=$(echo $PARAMS | tr [./] _)
-    TMP="${ROOT}/_tmp${TRANSLATED_PARAMS}"
-
-    # If we already have a checkout, update it, else make a new checkout.
-    if [ -d "${TMP}/.svn" ] ; then
-	svn update --quiet ${TMP}
-    else
-	rm -rf "${TMP}"
-	svn checkout --quiet "${SVN_URL}"/"${SVN_BRANCH}"/inputs "${TMP}"
-    fi
-
-    rm -f "${TMP}/cf_promises_release_id"
-
-    if /var/cfengine/bin/cf-promises -T "${TMP}"; then
-	md5sum `find ${TMP} -type f -name \*.cf` >"${TMP}/${CHECKSUM_FILE}"
-	if /usr/bin/diff -q "${TMP}/${CHECKSUM_FILE}" "${MASTERDIR}/${CHECKSUM_FILE}" ; then
-            # echo "No release needs to be made, the checksum files are the same"
-            touch "${TMP}"
-	else
-            cd "${TMP}" && (
-		chown -R root:root "${TMP}" && \
-                rsync -CrltDE -c --delete-after --chmod=u+rwX,go-rwx "${TMP}/" "${MASTERDIR}/" && \
-                rm -rf ${TMP}/.svn && \
-                echo "Successfully staged a policy release on $(date)"
-	    )
-	fi
-    else
-       error_exit "The staged policies in ${TMP} could not be validated, aborting."
-    fi
-}
+  STAGING_DIR="${ROOT}/_tmp${TRANSLATED_PARAMS}"
```

Comment:
agreed STAGING_DIR is a better name for the var.

But, I think that TRANSLATED_PARAMS needs to be set inside the vcs implementation.
I don't think the path to the params will be sufficent protection in all cases. For example I think if you changed from one repo to another you could have bad effects, so for a git implementation it might make more sense to include the git server and repo in that string that gets translated.

maybe instead of translated params a better name would be UPSTREAM_STAGING_DIR?

I am bad with names